api/web3server_test.go

```diff
@@ -21,50 +21,56 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+const (
+	_evmNetworkID uint32 = 1
+)
+
 func TestGetWeb3Reqs(t *testing.T) {
 	require := require.New(t)
-	testCase := []struct {
-		req      *http.Request
-		header   bool
-		hasError bool
+	testData := []struct {
+		testName  string
+		req       *http.Request
+		hasHeader bool
+		hasError  bool
 	}{
-		// fail to decode empty data
 		{
-			req:      httptest.NewRequest(http.MethodPost, "http://url.com", strings.NewReader("")),
-			header:   true,
-			hasError: true,
+			testName:  "EmptyData",
+			req:       httptest.NewRequest(http.MethodPost, "http://url.com", strings.NewReader("")),
+			hasHeader: true,
+			hasError:  true,
 		},
-		//  content-type is not json
 		{
-			req:      httptest.NewRequest(http.MethodPut, "http://url.com", strings.NewReader("")),
-			header:   false,
-			hasError: true,
+			testName:  "InvalidHttpMethod",
+			req:       httptest.NewRequest(http.MethodPut, "http://url.com", strings.NewReader("")),
+			hasHeader: false,
+			hasError:  true,
 		},
-		// missing params field
 		{
-			req:      httptest.NewRequest(http.MethodPost, "http://url.com", strings.NewReader(`{"jsonrpc":"2.0","method":"web3_clientVersion","id":67}`)),
-			header:   true,
-			hasError: false,
+			testName:  "MissingParamsField",
+			req:       httptest.NewRequest(http.MethodPost, "http://url.com", strings.NewReader(`{"jsonrpc":"2.0","method":"web3_clientVersion","id":67}`)),
+			hasHeader: true,
+			hasError:  false,
 		},
-		// success
 		{
-			req:      httptest.NewRequest(http.MethodPost, "http://url.com", strings.NewReader(`{"jsonrpc":"2.0","method":"web3_clientVersion","params":[],"id":67}`)),
-			header:   true,
-			hasError: false,
+			testName:  "Valid",
+			req:       httptest.NewRequest(http.MethodPost, "http://url.com", strings.NewReader(`{"jsonrpc":"2.0","method":"web3_clientVersion","params":[],"id":67}`)),
+			hasHeader: true,
+			hasError:  false,
 		},
 	}
 
-	for _, test := range testCase {
-		if test.header {
-			test.req.Header.Set("Content-Type", contentType)
-		}
-		_, err := parseWeb3Reqs(test.req)
-
-		if test.hasError {
-			require.Error(err)
-		} else {
-			require.NoError(err)
-		}
+	for _, test := range testData {
+		t.Run(test.testName, func(t *testing.T) {
```

Comment:
IMO we should not use `t.Run` here inside for loop? only couple of lines of code, not worth it

`t.Run` is meant to run the entire test with a different config, like
```
testIndexer := func(kvStore db.KVStore, t *testing.T) {
   // 100 lines of test code
}
t.Run("in memory", func(t *testing.T) {
	testIndexer(db.NewMemKVStore(), t)
})
t.Run("db files", func(t *testing.T) {
	testIndexer(db.NewBoltDB(), t)
})
```

Reply:
Different data is tested in the loop. With `t.run`, it's easier to trace where the err happens

Reply:
better keep the coding style consistency in test code